plugins/pulp_rpm/plugins/db/models.py

```diff
@@ -492,7 +492,132 @@ def download_path(self):
 
 
 class RpmBase(NonMetadataPackage):
-    # TODO add docstring to this class
+    """
+    This class is designed to be sub-classed by both RPM and SRPM, as these two
+    package types are similar. Most fields map to metadata fields in the RPM package
+    format.
+
+    :ivar arch: The target architecture for a package. For example, 'x86_64', 'i686',
+                or 'noarch'.
+    :type arch: mongoengine.StringField
+
+    :ivar buildhost: Hostname of the system that built the package.
+    :type buildhost: mongoengine.StringField
+
+    :ivar changelog: A list of changelog entries for the package. What these changelogs
+                     are used for depend on the packager, but Fedora uses to changelog
+                     to document changes to the package's spec file. The format of each
+                     entry is the time (in seconds since the epoch) of the changelog entry,
+                     the author and release as a string, and the list of release notes as
+                     a string.
+    :type changelog: list of [int, basestring, basestring]
+
+    :ivar checksum: The checksum of the package.
+    :type checksum: mongoengine.StringField
+
+    :ivar checksumtype: The checksum type used in the ``checksum`` field.
+    :type checksumtype: mongoengine.StringField
+
+    :ivar description: The human-readable description for a package.
+    :type description: mongoengine.StringField
+
+    :ivar epoch: The package's epoch.
+    :type epoch: mongoengine.StringField
+
+    :ivar files: All the files provided by this package. The dictionary has two keys: "file",
+                 and "dir". "file" maps to a list of strings, which are paths to files the
+                 package provides (for example, "/etc/pulp/server.conf". "dir" maps to a list
+                 of directories the package owns (for example, "/etc/pulp").
+    :type files: dict
+
+    :ivar group: The RPM group this package is a part of. As of ``rpm-4.13.0``,
+                 the following groups are documented in /usr/share/doc/rpm/GROUPS:
+                    Amusements/Games
```

Comment:
Does it make sense to link to the definition instead of copying contents from it into this docstring? I'm also fine to leave it as you've written it.